tools/GSDumpGUI/Forms/frmMain.cs

```diff
@@ -107,15 +116,61 @@ public GSDumpGUI()
 
             _availableGsDumps.OnIndexUpdatedEvent += UpdatePreviewImage;
 
-            txtGSDXDirectory.DataBindings.Add(nameof(TextBox.Text), Settings, nameof(Settings.GSDXDir));
-            txtDumpsDirectory.DataBindings.Add(nameof(TextBox.Text), Settings, nameof(Settings.DumpDir));
+            this.Text += " " + (IntPtr.Size * 8).ToString() + "bits";
+
+            if (String.IsNullOrEmpty(Settings.GSDXDir) || !Directory.Exists(Settings.GSDXDir))
+                Settings.GSDXDir = AppDomain.CurrentDomain.BaseDirectory;
+
+            if (String.IsNullOrEmpty(Settings.DumpDir) || !Directory.Exists(Settings.DumpDir))
+                Settings.DumpDir = AppDomain.CurrentDomain.BaseDirectory;
+
+            txtGSDXDirectory.Text = Settings.GSDXDir;
+            txtDumpsDirectory.Text = Settings.DumpDir;
 
             BindListControl(lstDumps, _availableGsDumps, g => g.Files, f => f.DisplayText, g => g.SelectedFileIndex);
             BindListControl(lstGSDX, _availableGsDlls, g => g.Files, f => f.DisplayText, g => g.SelectedFileIndex);
 
             Processes = new List<Process>();
 
             NoImage = CreateDefaultImage();
+
+            _dllWatcher = new List<FileSystemWatcher>();
+            _dumpWatcher = new List<FileSystemWatcher>();
+            _watcherEvents = new ConcurrentQueue<int>();
+
+            Thread watcherThread = new Thread(ChangesWatchdogThread);
+            watcherThread.Start();
+        }
+
+        private void ChangesWatchdogThread()
+        {
+            while (Program.isProgRunning)
+            {
+                bool dllReload = false;
+                bool dumpReload = false;
+
+                int evt;
+                while (_watcherEvents.TryDequeue(out evt))
+                {
+                    if (evt == 1) dllReload = true;
+                    else if (evt == 2) dumpReload = true;
+                }
+
+                if (dllReload) this.Invoke((MethodInvoker)delegate { ReloadGsdxDlls(); });
+                if (dumpReload) this.Invoke((MethodInvoker)delegate { ReloadGsdxDumps(); });
+
+                Thread.Sleep(1000);
```

Comment:
With other words a Timer with interval of 1s would be an alternative without that you manually need to handle threads.

Reply:
This one ? [System.Threading.Timer](https://docs.microsoft.com/en-us/dotnet/api/system.threading.timer) looks interesting. I'll give it a try.

Reply:
No better use the winforms timer to stay in the ui thread. You need to check whether it is disposable though and possibly dispose it as well on form dispose.

Reply:
https://docs.microsoft.com/de-de/dotnet/api/system.windows.forms.timer?view=netframework-4.8

Reply:
So the you could remove the this.Invoke and just call the update functions.

Reply:
Done! It seeems to be working alright.